link-grammar/tokenize/tokenize.c

```diff
@@ -3567,11 +3577,12 @@ bool flatten_wordgraph(Sentence sent, Parse_Options opts)
 				assert(!wpp_old->used, "Word %zu:%s has been used",
 				       wg_word->node_num, wpp_old->word->subword);
 
-				/* This is a new wordgraph word.
-				 */
+				/* This is a new wordgraph word.  */
 				assert(!right_wall_encountered, "Extra word");
-				if (!determine_word_expressions(sent, wg_word, &ZZZ_added, opts))
-					error_encountered = true;
+
+				wg_word->sent_wordidx = sent->length - 1;
+				gwappend(wa_word, wg_word);
```

Comment:
`gwordlist_append(&wa_word[wg_word->sent_wordidx], wg_word);`

Reply:
?

Reply:
This is a suggestion on how to use the existing `gwordlist_append()` (defined in `wordgraph.c`) instead of  `gwappend()` (because I recommended removing `gwappend()`).

Reply:
I just now pushed a patch to make this clearer.